plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHiveFileFormats.java

```diff
@@ -389,6 +393,54 @@ public void testIon(int rowCount, long fileSizePadding)
                 .isReadableByPageSource(fileSystemFactory -> new IonPageSourceFactory(fileSystemFactory, hiveConfig));
     }
 
+    @Test(dataProvider = "validRowAndFileSizePadding")
+    public void testIonWithTextEncoding(int rowCount, long fileSizePadding)
+            throws Exception
+    {
+        List<TestColumn> testColumns = TEST_COLUMNS.stream()
+                // todo: add support for maps to trino impl
+                .filter(tc -> !(tc.type instanceof MapType))
+                .collect(toList());
+
+        HiveConfig hiveConfig = new HiveConfig();
+        // enable Ion native trino integration for testing while the implementation is in progress
+        // TODO: In future this flag should change to `true` as default and then the following statement can be removed.
+        hiveConfig.setIonNativeTrinoEnabled(true);
+
+        assertThatFileFormat(ION)
+                .withColumns(testColumns)
+                .withRowsCount(rowCount)
+                .withFileSizePadding(fileSizePadding)
+                .withTableProperties(ImmutableMap.of(ION_ENCODING_PROPERTY, TEXT_ENCODING))
+                .withFileWriterFactory(fileSystemFactory -> new IonFileWriterFactory(fileSystemFactory, TESTING_TYPE_MANAGER))
+                .isReadableByPageSource(fileSystemFactory -> new IonPageSourceFactory(fileSystemFactory, hiveConfig));
+    }
+
+    @Test(dataProvider = "validRowAndFileSizePadding")
+    public void testInvalidIonEncoding(int rowCount, long fileSizePadding)
```

Comment:
This feels pretty "heavyweight" for what you're testing, a lot of setup, a lot of stuff that's extraneous for your test. I suspect the gap you found was in a test over the `IonFileWriterFactory` I would just create that now. We will need to test stuff like null field serialization there anyway.

Reply:
Could address this comment later though.

Reply:
Yes sounds good, I will add TODO and address this in the later PR.